### DISCUSSION TITLE
Support for new architecture without fabric?

### DISCUSSION THREAD

**Discussion by MateWW** [software-mansion/react-native-reanimated#4094] [Q&A]
Hey,
With react native 0.71 there is a really promising option to turn on turbo modules without enabling fabric.
Do you have the plan to support it somehow?

**Comment by tomekzaw** [comment]
Hey @MateWW, thanks for the question.

> there is a really promising option

Can you please describe what is the usecase for doing so (i.e. enabling TurboModules while still on Paper)? With NativeModules being mostly asynchronous, do you know if there are any libraries that would benefit from being a TurboModule, apart from type-safety and synchronicity? Have you encountered any problems when using Reanimated on such configuration? 

> Do you have the plan to support it somehow?

Reanimated 3.0.0-rc.0+ supports both renderers, Paper and Fabric. However, we use preprocessor conditionals to compile renderer-specific parts of code, so currently you cannot use both Paper and Fabric simultaneously within one RN app. The library automatically discovers whether `RCT_NEW_ARCH_ENABLED`/`newArchEnabled` flag is enabled and chooses which implementation to use. 

When it comes to TurboModules, indeed, `NativeReanimatedModuleSpec` extends `TurboModule` and has a `methodMap_`, but it doesn't really use codegen or autolinking. Instead, all the initialization is done either using JSI directly or via a single synchronous `install` method of a NativeModule, and NativeModules still work fine when TurboModules are enabled. Thanks to this unique architecture, Reanimated supports a wide range of React Native versions, from ~0.62 up to 0.71+, using the same code.
